sklearn/model_selection/_split.py

```diff
@@ -84,6 +84,12 @@ def split(self, X, y=None, groups=None):
 
         test : ndarray
             The testing set indices for that split.
+
+        Note
+        ----
+        Multiple calls to the ``split`` method will not return identical
+        training or testing sets if ``random_state`` parameter exists and is
```

Comment:
Maybe it's my lack of coffee, but why does ``split`` use ``_iter_test_masks``? Currently we create indices, transform them to booleans and then transform them back to indices. It is for making the negation easy?